### PR TITLE
add possobility to login with credentials

### PR DESCRIPTION
**Pull request by allomov**
I've write some code quickly to solve this problem. Currently I use [monkey patch](https://github.com/allomov/ilpomodoro/blob/ef7a498dea13efbce02eb634be3bb997b5a848f6/lib/ilpomodoro/tracker/todoist.rb#L8-L19) with this gem to add this functionality.

### CODE REVIEW

**Comment by namelessjon** [comment]
Oh, damn, I meant to get back to you about this sooner.

I don't get how this code is actually working?  It doesn't look much like the monkey patch you link to!

**Comment by namelessjon** [comment]
Oh wait, you added another patch, I missed that

File: lib/todoist/base.rb [hunk 1]
```diff
@@ -5,10 +5,24 @@ module Todoist
...
+    class AuthenticationError < StandardError; end 
+
     include HTTParty
     format :json
 
     ##
+    # Get authentication token using user email and password, 
+    # set up this token for making requests (see #setup method)
+    def self.login(email, password)
+      response = HTTParty.post('http://todoist.com/API/login', {body: {email: email, password: password}})
```
Comment (namelessjon) [inline lib/todoist/base.rb]: This needs to be to the https url.

**Comment by namelessjon** [comment]
Also, I no longer use the todoist service, so I don't have any possibility to test these changes myself.  Also, most importantly, thanks for your interest in the gem. :smile:

File: lib/todoist/base.rb (hunk 1 above)
Comment (allomov) [inline lib/todoist/base.rb]: sure, thanks, but it works as http also :)

**Comment by allomov** [comment]
@namelessjon what do you use now by the way? 
I tested this code for my [ilpomodoro](https://github.com/allomov/ilpomodoro/tree/add-todoist) branch. 

Your gem was very helpful for it, thank you.

File: lib/todoist/base.rb (hunk 1 above)
Comment (namelessjon) [inline lib/todoist/base.rb]: Then I think that's a problem with the todoist service.  But I don't see the point in unnecessarily exposing user credentials, even if the service will let you :wink:

**Comment by namelessjon** [comment]
I use orgmode for emacs, and sync changes about with git.  It works for what I need, TODOs wise, and means its integrated with the rest of my notes and journal.

**Pull request merged**